connectors/src/connectors/confluence/temporal/workflows.ts

```diff
@@ -0,0 +1,121 @@
+import type { ModelId } from "@dust-tt/types";
+import {
+  executeChild,
+  proxyActivities,
+  workflowInfo,
+} from "@temporalio/workflow";
+
+import type * as activities from "@connectors/connectors/confluence/temporal/activities";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+// The Temporal bundle does not support the use of aliases in import statements.
+import { makeConfluenceSpaceSyncWorkflowIdFromParentId } from "./utils";
+
+const {
+  confluenceGetSpaceNameActivity,
+  confluenceListPageIdsInSpaceActivity,
+  confluenceSaveStartSyncActivity,
+  confluenceUpsertPageActivity,
+  confluenceSaveSuccessSyncActivity,
+
+  fetchConfluenceConfigurationActivity,
+  getSpaceIdsToSyncActivity,
+} = proxyActivities<typeof activities>({
+  startToCloseTimeout: "20 minutes",
+});
+
+export async function confluenceFullSyncWorkflow({
+  connectionId,
+  connectorId,
+  dataSourceConfig,
+  spaceIdsToBrowse,
+}: {
+  connectionId: string;
+  connectorId: ModelId;
+  dataSourceConfig: DataSourceConfig;
+  spaceIdsToBrowse?: string[];
+}) {
+  await confluenceSaveStartSyncActivity(connectorId);
+
+  const spaceIdsToSync =
+    spaceIdsToBrowse ?? (await getSpaceIdsToSyncActivity(connectorId));
+
+  const {
+    workflowId,
+    searchAttributes: parentSearchAttributes,
+    memo,
+  } = workflowInfo();
+
+  for (const spaceId of spaceIdsToSync) {
+    await executeChild(confluenceSpaceSyncWorkflow, {
+      workflowId: makeConfluenceSpaceSyncWorkflowIdFromParentId(
+        workflowId,
+        spaceId
+      ),
+      searchAttributes: parentSearchAttributes,
+      args: [
+        {
+          connectionId,
+          connectorId,
+          dataSourceConfig,
+          isBatchSync: true,
+          spaceId,
+        },
+      ],
+      memo,
+    });
+  }
+
+  await confluenceSaveSuccessSyncActivity(connectorId);
+}
+
+interface ConfluenceSpaceSyncWorkflowInput {
+  connectionId: string;
+  connectorId: ModelId;
+  dataSourceConfig: DataSourceConfig;
+  isBatchSync: boolean;
+  spaceId: string;
+}
+
+export async function confluenceSpaceSyncWorkflow(
+  params: ConfluenceSpaceSyncWorkflowInput
+) {
+  const uniquePageIds = new Set<string>();
```

Comment:
Just so you know: Set are not properly serialized / deserialized by Temporal a runtime 
so if you end up passing it to an activity, you will have to manage that. No big deal.